### PR TITLE
HexChat macOS build fixes

### PR DESCRIPTION
**Pull request by ZachBacon**
so this should at least make it more buildable on macOS, there is still more fixes to be done (not entirely sure if my meson additions will be alright, but the issue I'm looking at atm is 

```
2018-06-27 04:24:44.934 hexchat[1804:12380] *** Terminating app due to uncaught exception 'NSImageCacheException', reason: 'Cannot lock focus on image <NSImage 0x7f97ad475f90 Size={0, 0} Reps=(
)>, because it is size zero.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff3613332b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff5d29dc76 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff361c4dcd +[NSException raise:format:] + 205
	3   AppKit                              0x00007fff337a2ca1 -[NSImage _lockFocusOnRepresentation:rect:context:hints:flipped:] + 316
	4   AppKit                              0x00007fff337a2b5e __51-[NSImage lockFocusWithRect:context:hints:flipped:]_block_invoke + 82
	5   AppKit                              0x00007fff337664d8 -[NSImage _usingBestRepresentationForRect:context:hints:body:] + 164
	6   AppKit                              0x00007fff337a2b03 -[NSImage lockFocusWithRect:context:hints:flipped:] + 162
	7   AppKit                              0x00007fff33806eda -[NSImage lockFocusFlipped:] + 123
	8   libgtkmacintegration-gtk2.2.dylib   0x0000000107417863 nsimage_from_pixbuf + 403
	9   libgtkmacintegration-gtk2.2.dylib   0x00000001074167d4 gtkosx_application_set_dock_icon_pixbuf + 84
	10  hexchat                             0x0000000106f0d68c fe_init + 44
	11  hexchat                             0x0000000106f4b3bd main + 349
	12  libdyld.dylib                       0x00007fff5deb7015 start + 1
	13  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Abort trap: 6
```
For now this is a good start in my opinion. I'll see about fixing one minor issue.

### CODE REVIEW

File: src/fe-gtk/meson.build [hunk 1]
```diff
@@ -50,7 +50,9 @@ hexchat_gtk_ldflags = [
 if get_option('with-libnotify')
   hexchat_gtk_sources += 'notifications/notification-libnotify.c'
   hexchat_gtk_deps += dependency('libnotify')
-elif false # TODO HAVE_GTK_MAC
+elif host_machine.system() == 'darwin'
+  hexchat_gtk_sources += 'notifications/notification-osx.m'
+  hexchat_gtk_deps += dependency('-framework Foundation')
```
Comment (TingPing) [inline src/fe-gtk/meson.build]: https://github.com/mesonbuild/meson/blob/dfd368d4054bfe067545112717a5bd21bc37aca7/test%20cases/objc/2%20nsstring/meson.build#L4

File: meson.build [hunk 2]
```diff
@@ -1,4 +1,4 @@
-project('hexchat', 'c',
+project('hexchat', 'c', 'objc',
```
Comment (TingPing) [inline meson.build]: `add_languages('objc')` conditional to being on osx.

File: meson.build [hunk 3]
```diff
@@ -31,6 +34,10 @@ config_h.set_quoted('LOCALEDIR', join_paths(get_option('prefix'),
                                  get_option('datadir'), 'locale'))
 config_h.set10('ENABLE_NLS', true)
 
+if host_machine.system() == 'darwin'
+  config_h.set('HAVE_GTK_MAC', true)
```
Comment (TingPing) [inline meson.build]: Could make this actually optional? `required: false` above and set value to `libgtk_mac_integration_dep.found()`

File: src/fe-gtk/meson.build [hunk 4]
```diff
@@ -31,6 +31,7 @@ hexchat_gtk_sources = [
 gtk_dep = dependency('gtk+-2.0', version: '>= 2.24.0')
 hexchat_gtk_deps = [
   hexchat_common_dep,
+  libgtk_mac_integration_dep, #used by gtk-mac-integration
```
Comment (TingPing) [inline src/fe-gtk/meson.build]: Redundant comment.

File: meson.build [hunk 5]
```diff
@@ -7,13 +7,16 @@ project('hexchat', 'c', 'objc',
...
+  add_languages('objc')
+endif
 
 i18n = import('i18n')
 gnome = import('gnome')
 cc = meson.get_compiler('c')
 
 if host_machine.system() == 'darwin'
-  libgtk_mac_integration_dep = dependency('gtk-mac-integration-gtk2')
+  libgtk_mac_integration_dep = dependency('gtk-mac-integration-gtk2', libgtk_mac_integration_dep.found())
```
Comment (TingPing) [inline meson.build]: Uh, you have these backwards, required goes here

File: src/fe-gtk/meson.build [hunk 6]
```diff
@@ -31,7 +31,6 @@ hexchat_gtk_sources = [
 gtk_dep = dependency('gtk+-2.0', version: '>= 2.24.0')
 hexchat_gtk_deps = [
   hexchat_common_dep,
-  libgtk_mac_integration_dep, #used by gtk-mac-integration
```
Comment (TingPing) [inline src/fe-gtk/meson.build]: You still have to pass the dep somewhere..

File: osx/hexchat.bundle [hunk 7]
```diff
@@ -22,30 +22,24 @@
...
 
   <binary>
     ${prefix}/lib/${gtkdir}/modules/*.so
   </binary>
   <binary>
-    ${prefix}/lib/${gtkdir}/${pkg:${gtk}:gtk_binary_version}/engines/libquartz.so
-  </binary>
-  <binary>
-    ${prefix}/lib/${gtkdir}/${pkg:${gtk}:gtk_binary_version}/engines/libxamarin.so
+    ${prefix}/lib/${gtkdir}/${pkg:${gtk}:gtk_binary_version}/engines/libmurrine.so
```
Comment (TingPing) [inline osx/hexchat.bundle]: Hmm, we do bundle a custom theme file that requires those engines.

File: src/fe-gtk/fe-gtk.c [hunk 8]
```diff
@@ -319,10 +319,11 @@ fe_init (void)
 	palette_load ();
 	key_init ();
 	pixmaps_init ();
-
+/*
```
Comment (TingPing) [inline src/fe-gtk/fe-gtk.c]: More information on whats failing here would be nice.

Comment (ZachBacon) [reply src/fe-gtk/fe-gtk.c]: It's more regarding to the error message on the top. For some reason it's causing an exception where the image is sized zero. Still kinda investigating this. But for now it's more of a temp fix.

**Comment by ZachBacon** [comment]
Ok, theme choice may not be as native, but I still feel it looks better. I'll let you decide, if you prefer the original I can revert.
screenshot

**Pull request closed without merging**